src/Domain/Amr/Helpers.hpp

```diff
@@ -74,4 +78,25 @@ template <size_t VolumeDim>
 std::vector<ElementId<VolumeDim>> ids_of_children(
     const ElementId<VolumeDim>& element_id,
     const std::array<Flag, VolumeDim>& flags);
+
+/// \ingroup AmrGroup
+/// \brief The ElementIds of the neighbors of `element` that will join with it
+/// given refinement `flags`
+///
+/// \note This function only returns the face neighbors of `element` that will
+/// join with it, and not the joining corner neighbors
+template <size_t VolumeDim>
+std::deque<ElementId<VolumeDim>> ids_of_joining_neighbors(
```

Comment:
Why `std::deque`?

Reply:
the action that uses it pops from the front and pushes to the back...